File: LeetCode Solutions/Problems/1221. Split a String in Balanced Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int result = 0, count = 0;
        for (int i=0; i<s.size(); i++){
            if (s[i] == 'L'){
                count--;
            }
            else{
                count++;
            }
            if (count == 0){
                result++;
            }
        }
        return result;
    }
};

File: LeetCode Solutions/Problems/1221. Split a String in Balanced Strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        result, count = 0, 0
        for i in range(len(s)):
            if s[i] == 'L':
                count -= 1
            else:
                count += 1
            if count == 0:
                result += 1
        return result